api/test/wfl/tools/datasets.clj

```diff
@@ -0,0 +1,51 @@
+(ns wfl.tools.datasets
+  (:require [clojure.string :as str]
+            [clojure.data.json :as json]
+            [wfl.service.datarepo :as datarepo]
+            [wfl.service.google.storage :as storage])
+  (:import (java.util UUID)))
+
+(defn unique-dataset-request
+  "Create a dataset request for a uniquely-named dataset defined by the json
+   file `dataset-basename` with the specified `tdr-profile`."
+  [tdr-profile dataset-basename]
+  (-> (str "test/resources/datasets/" dataset-basename)
+      slurp
+      json/read-str
+      ;; give it a unique name to avoid collisions with other tests
+      (update "name" #(str % (-> (UUID/randomUUID) (str/replace "-" ""))))
+      (update "defaultProfileId" (constantly tdr-profile))))
```

Comment:
Nit: with the specified `tdr-profile` -=> with `tdr-profile`?
What would an UNspecified one be?
Move the interior `;;` comment outside the `defn`?

Reply:
Moving the comment as the doc string explains this now.